applications/asset_tracker/src/cloud_codec/cloud_codec.c

```diff
@@ -83,3 +255,103 @@ int cloud_encode_sensor_data(const struct cloud_sensor_data *const sensor,
 
 	return 0;
 }
+
+static int cloud_search_cmd(cJSON *group_obj, enum cloud_cmd_group group)
+{
+	struct cmd *cmd_group;
+	cJSON *recipient_obj	= NULL;
+	cJSON *channel_obj	= NULL;
+	cJSON *type_obj		= NULL;
+
+	if (group_obj == NULL) {
+		return -EINVAL;
+	}
+
+	if (group == CLOUD_CMD_GROUP_SET) {
+		cmd_group = &group_set;
+	} else if (group == CLOUD_CMD_GROUP_GET) {
+		cmd_group = &group_get;
+	} else {
+		return -EINVAL;
+	}
+
+	for (size_t i = 0; i < cmd_group->num_children; i++) {
+		struct cmd rcpt = cmd_group->children[i];
+		recipient_obj = json_object_decode(group_obj,
+						   rcpt.name);
+		if (recipient_obj == NULL) {
+			continue;
+		}
+
+		cmd_parsed.recipient = rcpt.recipient;
+
+		for (size_t j = 0; j < rcpt.num_children; j++) {
+			struct cmd chan = rcpt.children[j];
+			channel_obj = json_object_decode(recipient_obj,
+							 chan.name);
+			if (channel_obj == NULL) {
+				continue;
+			}
+
+			cmd_parsed.channel = chan.channel;
+
+			for (size_t k = 0; k < chan.num_children; k++) {
+				struct cmd typ = chan.children[k];
+				type_obj = json_object_decode(channel_obj,
+							      typ.name);
+				if (type_obj == NULL) {
+					continue;
+				}
+
+				cmd_parsed.type = typ.type;
+
+				cloud_command_cb(&cmd_parsed);
+			}
+		}
+	}
+
+	return 0;
+}
+
+int cloud_decode_command(char const *input)
+{
+	int ret;
+	cJSON *root_obj		= NULL;
```

Comment:
Alignment.

Reply:
Alignment is OK on my end. Might be a github issue?

Reply:
My point was there's a huge gap between the variable name and the assignment operator, unnecessarily. Possibly because of the use of tabulation here.